## Setup

In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import scipy

## Data Overview

In [2]:
df = pd.read_csv("../data/raw/netflix_titles.csv")

In [3]:
df.rename(columns={"release_year":"year", "listed_in":"genre"}, inplace=True)

In [4]:
df.head()

,show_id,type,title,director,cast,country,date_added,year,rating,duration,genre,description
0,s1,Movie,Dick Johnson Is Dead,Kirsten Johnson,NaN,United States,"September 25, 2021",2020,PG-13,90 min,Documentaries,"As her father nears the end of his life, filmm..."
1,s2,TV Show,Blood & Water,NaN,"Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...",South Africa,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, TV Dramas, TV Mysteries","After crossing paths at a party, a Cape Town t..."
2,s3,TV Show,Ganglands,Julien Leclercq,"Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...",NaN,"September 24, 2021",2021,TV-MA,1 Season,"Crime TV Shows, International TV Shows, TV Act...",To protect his family from a powerful drug lor...
3,s4,TV Show,Jailbirds New Orleans,NaN,NaN,NaN,"September 24, 2021",2021,TV-MA,1 Season,"Docuseries, Reality TV","Feuds, flirtations and toilet talk go down amo..."
4,s5,TV Show,Kota Factory,NaN,"Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...",India,"September 24, 2021",2021,TV-MA,2 Seasons,"International TV Shows, Romantic TV Shows, TV ...",In a city of coaching centers known to train I...


In [5]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype
---  ------       --------------  -----
 0   show_id      8807 non-null   str  
 1   type         8807 non-null   str  
 2   title        8807 non-null   str  
 3   director     6173 non-null   str  
 4   cast         7982 non-null   str  
 5   country      7976 non-null   str  
 6   date_added   8797 non-null   str  
 7   year         8807 non-null   int64
 8   rating       8803 non-null   str  
 9   duration     8804 non-null   str  
 10  genre        8807 non-null   str  
 11  description  8807 non-null   str  
dtypes: int64(1), str(11)
memory usage: 825.8 KB


In [6]:
df.nunique()

show_id        8807
type              2
title          8807
director       4528
cast           7692
country         748
date_added     1767
year             74
rating           17
duration        220
genre           514
description    8775
dtype: int64

The dataset includes 8,807 Netflix titles across 12 columns, with both Movies and TV Shows represented. It provides the main information needed for this analysis, including content type, release year, rating, duration, genre, country, and date added.

The first inspection suggests that the core analysis columns are in good condition overall. The main missing-value issues appear in `director`, `cast`, and `country`, which means these fields should be handled carefully before using them for deeper analysis.